wandb/sdk/internal/file_stream.py

```diff
@@ -8,31 +8,64 @@
 import sys
 import threading
 import time
-from typing import Any, Dict, List, Optional, Tuple
+from types import TracebackType
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    NamedTuple,
+    Optional,
+    Set,
+    Tuple,
+    Type,
+    TYPE_CHECKING,
+    Union,
+)
+
+if TYPE_CHECKING:
+    if sys.version_info >= (3, 8):
+        from typing import TypedDict
+    else:
+        from typing_extensions import TypedDict
+
+    class ProcessedChunk(TypedDict):
+        offset: int
+        content: List[str]
+
+    class ProcessedBinaryChunk(TypedDict):
+        offset: int
+        content: str
+        encoding: str
+
 
 import requests
 import wandb
 from wandb import env, util
+from wandb.sdk.internal import internal_api, settings_static
 
 from ..lib import file_stream_utils
 
 logger = logging.getLogger(__name__)
 
-Chunk = collections.namedtuple("Chunk", ("filename", "data"))
+
+class Chunk(NamedTuple):
+    filename: str
+    data: Any
 
 
 class DefaultFilePolicy:
-    def __init__(self, start_chunk_id=0):
+    def __init__(self, start_chunk_id: int = 0) -> None:
         self._chunk_id = start_chunk_id
 
-    def process_chunks(self, chunks):
+    def process_chunks(self, chunks: List[Chunk]) -> "ProcessedChunk":
```

Comment:
If you change the return type of this method to `Union[bool, "ProcessedChunk", "ProcessedBinaryChunk", List["ProcessedChunk"]]`, I believe the `# type: ignore`s in the subclass overrides become unnecessary!

Reply:
done, thanks!